src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -397,6 +706,21 @@ private BoundExpression ConvertConcatExprToString(SyntaxNode syntax, BoundExpres
                 }
             }
 
+            // If expression is of form `constantChar.ToString()` then rewrite it here to a string literal so we can choose better options for lowering later (e.g. fold it with another constant instead of performing concatenation)
+            // NOTE: We request `object.ToString()` as an optional member here to avoid reporting "missing member" diagnostic earlier than we should.
+            // Because if in the end it turns out that we have a simple concatenation of n strings there is no need to report missing `object.ToString()` member
+            if (TryGetSpecialTypeMethod(expr.Syntax, SpecialMember.System_Object__ToString, out MethodSymbol? optionalObjectToString, isOptional: true) &&
```

Comment:
>TryGetSpecialTypeMethod(expr.Syntax, SpecialMember.System_Object__ToString, out MethodSymbol? optionalObjectToString, isOptional: true) && [](http://example.com/codeflow?start=16&length=138)

I suggest to reorder and add some conditions to this `if`, with the goal of performing cheaper checks first, and filtering out expressions that are not going to match faster.
Here is the set of conditions that I have in mind:
1. Check if type of the expression is string
2. Check that it is a call of a method named "ToString" on a char constant
3. Check that the method matches the method we are looking for, requesting the ```SpecialMember.System_Object__ToString``` in the process.